src/main/java/Duke.java

```diff
@@ -1,10 +1,83 @@
+import java.util.Scanner;
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        DukePrinter.printGreeting();
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            //todo not follow SLAP here, fixed it later.
+            String line = in.nextLine();
+            line = line.trim();
+            String[] command = line.split(" ");
+            String lineRemaining = line.substring(line.indexOf(" ")+1);
+            lineRemaining = lineRemaining.trim();
+            int id;
+            switch (command[0]){
+            case "bye":
+                DukePrinter.printStringln("Bye. Hope to see you again soon!");
+                DukePrinter.printGoodbyeLogo();
+                return;
+            case "list":
+                DukeList.listTask();
+                break;
+            case "mark":
+                try {
+                    id = Integer.parseInt(lineRemaining);
+                } catch (NumberFormatException integerException) {
+                      DukePrinter.printErrorln("Sorry, the id is invalid!");
+                    break;
+                }
+                DukeList.markDone(id-1);
+                break;
+            case "unmark":
+                try {
+                    id = Integer.parseInt(lineRemaining);
```

Comment:
Since you have already parsed the input into a string array, wouldn't it be better to use the second index of the  command array instead?